src/UnsetDataCapableTrait.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Dhii\Data\Object;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+
+/**
+ * Functionality for data retrieval.
+ *
+ * @since [*next-version*]
+ */
+trait UnsetDataCapableTrait
+{
+    /**
+     * Unset data by keys.
+     *
+     * @since [*next-version*]
+     *
+     * @param Stringable[]|Traversable $keys The keys of data to unset.
+     */
+    protected function _unsetData($keys)
+    {
+        if (!is_array($keys) && !($keys instanceof Traversable)) {
+            throw $this->_createInvalidArgumentException($this->__('Keys must be iterable'), null, null, $keys);
+        }
+
+        $store = $this->_getDataStore();
+        foreach ($keys as $_idx => $_key) {
+            if (!is_string($_key) && !($_key instanceof Stringable)) {
+                throw $this->_createInvalidArgumentException($this->__('Data key #%1$s must be stringable', $_idx), null, null, $keys);
+            }
+
+            unset($store->{(string) $_key});
+        }
+    }
+
+    /**
+     * Retrieves a pointer to the data store.
+     *
+     * @since [*next-version*]
+     *
+     * @return mixed The data store.
```

Comment:
Perhaps this should type-hint a return type of `object` since `_unsetData()` expect an object.